src/Cardano/Wallet/Binary.hs

```diff
@@ -0,0 +1,368 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE RankNTypes #-}
+{-# OPTIONS_GHC -fno-warn-name-shadowing #-}
+
+-- | These are (partial) CBOR decoders for blocks and block headers.
+-- Note that we ignore most of the block's and header's content and only
+-- retrieve the pieces of information relevant to us, wallet (we do assume a
+-- trusted node and therefore, we needn't to care about verifying signatures and
+-- blocks themselves).
+
+module Cardano.Wallet.Binary
+    ( decodeBlock
+    , decodeBlockHeader
+    ) where
+
+import qualified Codec.CBOR.Decoding as CBOR
+import qualified Codec.CBOR.Encoding as CBOR
+import qualified Codec.CBOR.Read as CBOR
+import qualified Codec.CBOR.Write as CBOR
+import Control.Monad
+    ( void )
+import qualified Data.ByteString.Lazy as BL
+import Data.Set
+    ( Set )
+import qualified Data.Set as Set
+import Data.Text
+    ( Text )
+import Data.Word
+    ( Word16, Word64 )
+import Prelude
+
+import Cardano.Wallet.Binary.Helpers
+import Cardano.Wallet.Primitive
```

Comment:
May we avoid implicit imports (except for `Prelude`) :pray:

Reply:
This seems excessive to me, but I have changed it.